File: src/tasks/cli/feature/info/InfoParser.java
package tasks.cli.feature.info;

import omnia.data.cache.Memoized;
import omnia.data.structure.List;
import tasks.cli.command.common.simple.SimpleParser;
import tasks.cli.parser.ParseResult;
import tasks.cli.parser.Parser;
import tasks.model.ObservableTask;

/** Command line argument parser for the Info command. */
public final class InfoParser extends SimpleParser<InfoArguments> {
  public InfoParser(
      Memoized<? extends Parser<? extends List<? extends ParseResult<? extends ObservableTask>>>>
          taskParser) {
    super(InfoArguments::new, taskParser);
  }
}

File: cli/src/me/strangepan/tasks/cli/feature/info/InfoParser.kt
package me.strangepan.tasks.cli.feature.info

import java.util.function.Function
import omnia.data.cache.Memoized
import omnia.data.structure.List
import me.strangepan.tasks.cli.command.common.simple.SimpleParser
import me.strangepan.tasks.cli.parser.ParseResult
import me.strangepan.tasks.cli.parser.Parser
import me.strangepan.tasks.engine.model.Task

/** Command line argument parser for the Info command.  */
class InfoParser(
    taskParser: Memoized<out Parser<out List<out ParseResult<out Task>>>>)
    : SimpleParser<InfoArguments>(Function(::InfoArguments), taskParser)